service-base/src/main/java/org/eclipse/hono/service/VertxBasedHealthCheckServer.java

```diff
@@ -159,18 +161,20 @@ public void setAdditionalResources(final List<Handler<Router>> resourceProviders
                     result.fail(startAttempt.cause());
                 }
             });
-            return result;
-        } else {
-            return Future.succeededFuture();
         }
+            return result;
     }
 
     private Future<Void> bindSecureHttpServer() {
-        if (config.isSecurePortEnabled()) {
+        if (config.getBindAddress() == Constants.LOOPBACK_DEVICE_ADDRESS
+        || ( config.getBindAddress() == Constants.LOOPBACK_DEVICE_ADDRESS && config.getPort() == Constants.PORT_UNCONFIGURED )) {
```

Comment:
IMHO it should be sufficient to check if a bind address has been set, shouldn't it?

Reply:
ah good catch, there is no need to check the port here